### PR TITLE
Create mongodb index by region to speed up signal queries

### PR DESCRIPTION
**Pull request by leexgh**
Fix: https://github.com/cBioPortal/cbioportal/issues/8318

Build signal query index by `chromosome`, `start` and `end`. 
10k queries take more than 10 minutes before, and take about 30s now.

### CODE REVIEW

File: service/src/main/java/org/cbioportal/genome_nexus/service/enricher/SignalAnnotationEnricher.java [hunk 1]
```diff
@@ -23,13 +24,9 @@ public SignalAnnotationEnricher(
...
     public void enrich(VariantAnnotation annotation) {
         SignalAnnotation signalAnnotation = new SignalAnnotation();
-        String hgvsg = annotation.getHgvsg();
-
-        if (hgvsg != null) {
-            List<SignalMutation> mutations = this.signalMutationService.getSignalMutationsByHgvsg(hgvsg);
-            signalAnnotation.setAnnotation(mutations);
-        }
-
+        GenomicLocation genomicLocation = GenomicLocationResolver.resolve(annotation);
```
Comment (onursumer) [inline service/src/main/java/org/cbioportal/genome_nexus/service/enricher/SignalAnnotationEnricher.java:27]: Let's revert the changes in `GenomicLocationResolver` and create an instance of `GenomicLocationResolver` in the constructor instead. Something like this should work:

```Java
private final GenomicLocationResolver genomicLocationResolver;

public SignalAnnotationEnricher(...) {
    ...
    this.genomicLocationResolver = new GenomicLocationResolver()
}

```

Then you can simply do this here:
```suggestion
        GenomicLocation genomicLocation = this.genomicLocationResolver.resolve(annotation);
```

**Review by onursumer — CHANGES_REQUESTED** [review]

File: service/src/main/java/org/cbioportal/genome_nexus/service/enricher/SignalAnnotationEnricher.java (hunk 1 above)
Comment (leexgh) [reply service/src/main/java/org/cbioportal/genome_nexus/service/enricher/SignalAnnotationEnricher.java]: @onursumer Sure, updated. Why static is not good here?

Comment (onursumer) [reply service/src/main/java/org/cbioportal/genome_nexus/service/enricher/SignalAnnotationEnricher.java]: Thanks! Static methods may break component autowiring. Also, I think only dynamic methods can be mocked in test classes.

**Review by onursumer — APPROVED** [review]
LGTM
